src/main/java/com/paloaltonetworks/panorama/api/methods/PanoramaApiClient.java

```diff
@@ -0,0 +1,575 @@
+/* Copyright 2016 Palo Alto Networks Inc.
+ * All Rights Reserved.
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+package com.paloaltonetworks.panorama.api.methods;
+
+import static java.util.Collections.emptyList;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.stream.Collectors.toList;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Invocation.Builder;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.UriBuilder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.paloaltonetworks.panorama.api.mapping.AddressEntry;
+import com.paloaltonetworks.panorama.api.mapping.CommitResponse;
+import com.paloaltonetworks.panorama.api.mapping.DAGResponse;
+import com.paloaltonetworks.panorama.api.mapping.DeviceEntry;
+import com.paloaltonetworks.panorama.api.mapping.DeviceGroupResponse;
+import com.paloaltonetworks.panorama.api.mapping.DeviceGroupsEntry;
+import com.paloaltonetworks.panorama.api.mapping.GetAddressResponse;
+import com.paloaltonetworks.panorama.api.mapping.GetTagResponse;
+import com.paloaltonetworks.panorama.api.mapping.PANResponse;
+import com.paloaltonetworks.panorama.api.mapping.SetConfigResponse;
+import com.paloaltonetworks.panorama.api.mapping.ShowDeviceResponse;
+import com.paloaltonetworks.panorama.api.mapping.ShowResponse;
+import com.paloaltonetworks.panorama.api.mapping.TagEntry;
+import com.paloaltonetworks.panorama.api.mapping.VMAuthKeyResponse;
+import com.paloaltonetworks.utils.QuickXmlizerUtil;
+
+public class PanoramaApiClient {

```

Comment:
can we split this into more functional pieces?

Maybe a authentication/Commit API, device group api etc?